yext/answers-headless-react
# SessionId support

- update AnswersHeadlessProvider to accept a sessionTrackingEnabled field. If true, either 1) get session id from local storage or 2) generate a uuid and store that in local storage. Set the id as sessionId in headless state.

J=SLAP-1726
TEST=manual

- clear all cookies, set sessionTrackingEnabled to true. See that sessionId is tagged in universal and vertical search request. set sessionTrackingEnabled to false and see that sessionId no longer appears
- set sessionTrackingEnabled to true and commented out setSessionId. see that cookie is dropped.
- set sessionTrackingEnabled to true, see that session id is the same in the same tab over reloads. Opening a new tab creates a new session id.
